lib/model/store.dart

```diff
@@ -441,14 +487,20 @@ class PerAccountStore extends ChangeNotifier with EmojiStore, ChannelStore, Mess
 
   @override
   void dispose() {
+    assert(!_disposed);
     recentDmConversationsView.dispose();
     unreads.dispose();
     _messages.dispose();
     typingStatus.dispose();
+    updateMachine?.dispose();
+    connection.close();
```

Comment:
nit:

    This seems hygienic. It makes sense to explicitly clean up HTTP
    client resources when we know we'll be setting up a new client
    instance to replace the old one. (The dead-queue reload is currently
    the only path where PerAccountStore.dispose is called.)

In this situation it's not that we will be setting up a new HTTP client instance; rather we already have, which was passed to the new PerAccountStore (and even before that, was used for fetching the new store's initial snapshot). The new store gets constructed before the old one is disposed.